openwisp_controller/geo/api/views.py

```diff
@@ -123,10 +123,19 @@ def get_queryset(self):
         except ValidationError:
             return qs.none()
 
+    def initial(self, *args, **kwargs):
+        super().initial(*args, **kwargs)
+        self.assert_parent_exists()
+
+    def assert_parent_exists(self):
```

Comment:
#543  logic seems redundant with the reusable `FilterByParentManaged` which we have in openwisp-users, see [Checking parent objects](https://openwisp.io/docs/dev/users/developer/django-rest-framework-utils.html#checking-parent-objects). Can you try reusing that logic to avoid duplication to see if it fixes the issue?

If any improvement is needed in the reusable class and can be done without causing issues to the rest of the codebase, I am more inclined to puruse that path so we maintain consistency across the codebase, otherwise future maintenance will be insane.